### ISSUE TITLE
Add formatting guide (aka clang-format config)

### ISSUE THREAD

**Issue by hpwxf** [RUrlus/carma#34]
To ensure a uniform code style, I suggest adding a `.clang-format` config file.

I can try to provide a config close to the current coding style.

**Comment by RUrlus** [comment]
Sounds good, only strong preferences I have are:

- 4 space indent
- opening on same line, close on new line, e.g.
```C++
if (something) {
    do_something_here
}
```

Indent within namespaces I am fine with either indent or no indent

**Comment by hpwxf** [comment]
Ok, understood 👍

**Comment by hpwxf** [comment]
I suggest this configuration:
[clang-format.txt](https://github.com/RUrlus/carma/files/4886759/clang-format.txt)

You can try it here: https://zed0.co.uk/clang-format-configurator/

**Comment by RUrlus** [comment]
Only thing I don't like about the config is:
```
std::vector<uint32_t> returnVector( uint32_t* LongNameForParameter1,
                                    double* LongNameForParameter2,
                                    const float& LongNameForParameter3,
                                    const std::map<std::string,int32_t>& LongNameForParameter4) {
```

Preferring:

```C++
std::vector<uint32_t> returnVector(
    uint32_t* LongNameForParameter1,
    double* LongNameForParameter2,
    const float& LongNameForParameter3,
    const std::map<std::string, int32_t>& LongNameForParameter4
) {
```

However this doesn't seem possible, the closest I got:

```C++
std::vector<uint32_t> returnVector(
    uint32_t* LongNameForParameter1,
    double* LongNameForParameter2,
    const float& LongNameForParameter3,
    const std::map<std::string, int32_t>& LongNameForParameter4) {
```

What are your thoughts?

config:
```
---
BasedOnStyle: Chromium
AllowShortBlocksOnASingleLine: 'false'
AllowShortCaseLabelsOnASingleLine: 'false'
AllowShortIfStatementsOnASingleLine: 'false'
AllowShortLoopsOnASingleLine: 'false'
AlwaysBreakTemplateDeclarations: 'Yes'
BinPackArguments: 'false'
BinPackParameters: 'false'
BreakBeforeBinaryOperators: All
BreakInheritanceList: BeforeComma
ColumnLimit: '120'
CompactNamespaces: 'false'
FixNamespaceComments: 'true'
IndentWidth: 4
UseTab: Never
ObjCBlockIndentWidth: 4
ContinuationIndentWidth: 4
AlignAfterOpenBracket: AlwaysBreak
IndentCaseLabels: true
AllowAllParametersOfDeclarationOnNextLine: false
```

**Comment by hpwxf** [comment]
Except `ObjCBlockIndentWidth` which is for Objective-C code, I've updated clang-format following your config (cf #37).
(and I don't know how to do a style closer to your preferred style).

**Comment by RUrlus** [comment]
Thanks for the work, closed by #37
